File: stack/682-Baseball-Game.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        ans = []
        for operation in operations:
            if operation.isdigit() or operation.lstrip('-').isdigit():
                ans.append(int(operation))
            elif operation.isalpha() and operation == 'D':
                current_score = ans[-1] * 2
                ans.append(current_score)
            elif operation.isalpha() and operation == 'C':
                ans.pop()
            else:
                if len(ans) >= 2:
                    current_score = ans[-1] + ans[-2]
                    ans.append(current_score)
        
        return 0 if len(ans) == 0 else sum(ans)

File: stack/682-Baseball-Game.go
func calPoints(operations []string) int {
    var ans[]int

    for _, operation := range operations {
        if num, err := strconv.Atoi(operation); err == nil {
            ans = append(ans, num)
        } else if operation == \D\ && len(ans) > 0 {
            currentScore := ans[len(ans) - 1] * 2
            ans = append(ans, currentScore)
        } else if operation == \C\ && len(ans) > 0 {
            ans = ans[:len(ans) - 1]
        } else if operation == \+\ && len(ans) >= 2 {
            currentScore := ans[len(ans) - 1] + ans[len(ans) - 2]
            ans = append(ans, currentScore)
        }
    }

    total := 0
    for _, score := range ans {
        total += score
    }
    return total
}